core/trino-main/src/test/java/io/trino/sql/planner/TestLogicalPlanner.java

```diff
@@ -1332,6 +1332,32 @@ public void testRemovesTrivialFilters()
                         values("nationkey", "name", "regionkey", "comment")));
     }
 
+    @Test
+    public void testRemoveTrivialFiltersWithPartitioningHandle()
+    {
+        assertThat(countOfMatchingNodes(
+                plan("""
+                    SELECT *
```

Comment:
Is this the smallest query that reproduces the issue?

Reply:
This query needs to retain these features to trigger the specific sequence of optimizer rules, and it has been simplified as much as possible.